src/Enzo/enzo_EnzoMethodMHDVlct.cpp

```diff
@@ -36,20 +36,31 @@ static str_vec_t concat_str_vec_(const str_vec_t& vec1, const str_vec_t& vec2){
 EnzoMethodMHDVlct::EnzoMethodMHDVlct (std::string rsolver,
 				      std::string half_recon_name,
 				      std::string full_recon_name,
-				      double gamma, double theta_limiter,
-				      double density_floor,
-				      double pressure_floor,
+				      double theta_limiter,
 				      std::string mhd_choice,
-				      bool dual_energy_formalism,
-				      double dual_energy_formalism_eta,
 				      bool store_fluxes_for_corrections)
   : Method()
 {
+  // check compatability with EnzoPhysicsFluidProps
+  EnzoPhysicsFluidProps* fluid_props = enzo::fluid_props();
+  const EnzoDualEnergyConfig& de_config = fluid_props->dual_energy_config();
+  ASSERT("EnzoMethodMHDVlct::EnzoMethodMHDVlct",
+         "selected formulation of dual energy formalism is incompatible",
+         de_config.is_disabled() | de_config.modern_formulation());
```

Comment:
Do you really want to use the bitwise | operator here (why not ||)? Feel free to ignore if I am missing something.

Reply:
I'll change this since it's confusing here. 

This is a weird habit I've picked up (I've started to use the bitwise `&`  and bitwise `|` operators instead of `&&` and `||` unless I really need the short-circuiting functionality. But, it would be better to use the logical operators by-default)